### PR TITLE
dRNG refactor

### PR DESCRIPTION
**Pull request by capossele**

### CODE REVIEW

File: packages/drng/collective_beacon_payload.go [hunk 1]
```diff
@@ -109,7 +106,7 @@ func FromBytes(bytes []byte, optionalTargetObject ...*Payload) (result *Payload,
 }
 
 // Bytes returns the collective beacon payload bytes.
-func (payload *Payload) Bytes() (bytes []byte) {
+func (payload *CollectiveBeaconPayload) Bytes() (bytes []byte) {
```
Comment (jkrvivian) [inline packages/drng/collective_beacon_payload.go:109]: Rename the method receiver to a single character. Like this:
```suggestion
func (p *CollectiveBeaconPayload) Bytes() (bytes []byte) {
```

File: packages/drng/common.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package payload
+package drng
```
Comment (jkrvivian) [inline packages/drng/common.go:1]: Maybe rename the file to `constant.go`?

File: packages/drng/dispatcher.go [hunk 3]
```diff
@@ -4,27 +4,22 @@ import (
...
-	cb "github.com/iotaledger/goshimmer/packages/binary/drng/subtypes/collectivebeacon/payload"
 	"github.com/iotaledger/hive.go/crypto/ed25519"
 	"github.com/iotaledger/hive.go/marshalutil"
 )
 
 // Dispatch parses a DRNG message and process it based on its subtype
-func (drng *DRNG) Dispatch(issuer ed25519.PublicKey, timestamp time.Time, payload *payload.Payload) error {
-	switch payload.Header.PayloadType {
-	case header.TypeCollectiveBeacon:
+func (drng *DRNG) Dispatch(issuer ed25519.PublicKey, timestamp time.Time, payload *Payload) error {
```
Comment (jkrvivian) [inline packages/drng/dispatcher.go:12]: Single character for method receiver here.
```suggestion
func (d *DRNG) Dispatch(issuer ed25519.PublicKey, timestamp time.Time, payload *Payload) error {
```

File: packages/drng/events.go [hunk 4]
```diff
@@ -16,7 +17,7 @@ type CollectiveBeaconEvent struct {
 	InstanceID uint32
 	// Round of the current beacon.
 	Round uint64
-	// Collective signature of the previous beacon.
+	// Collective signature of the previous beacon.s
```
Comment (jkrvivian) [inline packages/drng/events.go:20]: Typo.
```suggestion
	// Collective signature of the previous beacons.
```

File: packages/drng/payload.go [hunk 5]
```diff
@@ -83,64 +82,64 @@ func FromBytes(bytes []byte) (result *Payload, consumedBytes int, err error) {
...
 
 // region Payload implementation ///////////////////////////////////////////////////////////////////////////////////////
 
-// Type defines the type of the drng payload.
-var Type = tangle.PayloadType(111)
+// PayloadType defines the type of the drng payload.
+var PayloadType = tangle.PayloadType(111)
 
 // Type returns the type of the drng payload.
 func (payload *Payload) Type() tangle.PayloadType {
```
Comment (jonastheis) [inline packages/drng/payload.go:132]:
```suggestion
func (p *Payload) Type() tangle.PayloadType {
```

**Review by jonastheis — CHANGES_REQUESTED** [review]

**Review by jonastheis — APPROVED** [review]

**Pull request merged**